src/main/java/org/apache/sling/feature/cpconverter/cli/ContentPackage2FeatureModelConverterLauncher.java

```diff
@@ -101,6 +101,9 @@
     @Option(names = { "--enforce-principal-based-supported-path" }, description = "Converts service user access control entries to principal-based setup using the given supported path.", required = false)
     private String enforcePrincipalBasedSupportedPath = null;
 
+    @Option(names = { "--enforce-servicemapping-by-principal" }, description = "Converts service user mappings with the form 'service:sub=userID' to 'service:sub=[principalname]'.", required = false)
```

Comment:
It should be clarified how the mapping is done: To which degree are the user's group membership considered?

Reply:
good point. i will expand the comment explaining that group membership will potentially no longer be resolved. the reason why i say "potentially": it depends on the implementation. i am not too familiar with the default in sling but with AEM group membership is not resolved as this was the primary reason for introducing the mapping by principal-names. group membership with service users should be considered a potential security risk because permission setup with service users is part of the application/service, while permissions for groups in particular everyone is not part of the application/service any may vary from installation to installation.

Reply:
Sling behaves the same as AEM: https://github.com/apache/sling-org-apache-sling-jcr-base/blob/66be360910c265473799635fcac0e23895898913/src/main/java/org/apache/sling/jcr/base/AbstractSlingRepository2.java#L240

Reply:
@kwin , group principals are not valid in service user mapping.... so yes this is on purpose.